### PR TITLE
fix: rename redis.conf and replace the word redis to valkey

### PR DESCRIPTION
**Pull request by davidyu62**
rename redis.conf to valkey.conf and replace all redis with valkey in the file.

### CODE REVIEW

File: redis.conf [hunk 1]
```diff
@@ -173,7 +173,7 @@ timeout 0
 # On other kernels the period depends on the kernel configuration.
 #
 # A reasonable value for this option is 300 seconds, which is the new
-# Redis default starting with Redis 3.2.1.
```
Comment (hwware) [inline redis.conf:176]: There is no Valkey 3.2.1 version, I prefer to keep the Redis 3.2.1 or indicate it origins from Redis 3.2.1

File: redis.conf [hunk 2]
```diff
@@ -540,7 +540,7 @@ dir ./
 #
 # masterauth <master-password>
 #
-# However this is not enough if you are using Redis ACLs (for Redis version
```
Comment (hwware) [inline redis.conf:543]: Valkey first version is 7.2, here, we could write it comes from the Redis 6 or greater or Valkey 7.2

File: redis.conf [hunk 3]
```diff
@@ -573,7 +573,7 @@ replica-serve-stale-data yes
 # may also cause problems if clients are writing to it because of a
 # misconfiguration.
 #
-# Since Redis 2.6 by default replicas are read-only.
```
Comment (hwware) [inline redis.conf:576]: Same as above, there is no Valkey 2.6

File: redis.conf [hunk 4]
```diff
@@ -1029,11 +1029,11 @@ acllog-max-len 128
 # ACL file, the server will refuse to start.
 #
 # The format of the external ACL user file is exactly the same as the
-# format that is used inside redis.conf to describe users.
+# format that is used inside valkey.conf to describe users.
 #
-# aclfile /etc/redis/users.acl
+# aclfile /etc/valkey/users.acl
 
-# IMPORTANT NOTE: starting with Redis 6 "requirepass" is just a compatibility
```
Comment (hwware) [inline redis.conf:1036]: Same as above, there is no Valkey 6 version, pls update this information properly.

File: redis.conf [hunk 5]
```diff
@@ -1045,15 +1045,15 @@ acllog-max-len 128
 # requirepass foobared
 
 # New users are initialized with restrictive permissions by default, via the
-# equivalent of this ACL rule 'off resetkeys -@all'. Starting with Redis 6.2, it
```
Comment (hwware) [inline redis.conf:1048]: Same as above

File: redis.conf [hunk 6]
```diff
@@ -1045,15 +1045,15 @@ acllog-max-len 128
...
+# equivalent of this ACL rule 'off resetkeys -@all'. Starting with Valkey 6.2, it
 # is possible to manage access to Pub/Sub channels with ACL rules as well. The
 # default Pub/Sub channels permission if new users is controlled by the
 # acl-pubsub-default configuration directive, which accepts one of these values:
 #
 # allchannels: grants access to all Pub/Sub channels
 # resetchannels: revokes access to all Pub/Sub channels
 #
-# From Redis 7.0, acl-pubsub-default defaults to 'resetchannels' permission.
```
Comment (hwware) [inline redis.conf:1056]: Same as above

File: redis.conf [hunk 7]
```diff
@@ -1368,33 +1368,33 @@ disable-thp yes
...
 # by setting it via CONFIG command on a live server first.
 #
-# Please check https://redis.io/topics/persistence for more information.
+# Please check https://valkey.io/topics/persistence for more information.
 
 appendonly no
 
 # The base name of the append only file.
 #
-# Redis 7 and newer use a set of append-only files to persist the dataset
```
Comment (hwware) [inline redis.conf:1397]: Same as above, perfer to mention Redis 7 and greater, and Valkey 7.2 or similar words

File: redis.conf [hunk 8]
```diff
@@ -1444,7 +1444,7 @@ appenddirname "appendonlydir"
 # everysec.
 #
 # More details please check the following article:
-# http://antirez.com/post/redis-persistence-demystified.html
```
Comment (hwware) [inline redis.conf:1447]: Here, we should keep redis

File: redis.conf [hunk 9]
```diff
@@ -2226,19 +2226,19 @@ rdb-save-incremental-fsync yes
...
-# Active (online) defragmentation allows a Redis server to compact the
+# Active (online) defragmentation allows a Valkey server to compact the
 # spaces left between small allocations and deallocations of data in memory,
 # thus allowing to reclaim back memory.
 #
 # Fragmentation is a natural process that happens with every allocator (but
 # less so with Jemalloc, fortunately) and certain workloads. Normally a server
 # restart is needed in order to lower the fragmentation, or at least to flush
 # away all the data and create it again. However thanks to this feature
-# implemented by Oran Agra for Redis 4.0 this process can happen at runtime
```
Comment (hwware) [inline redis.conf:2237]: Pls update properly as above

File: redis.conf (hunk 3 above)
Comment (daniel-house) [reply redis.conf:576]: Do we have an agreed approach to versioning and referring to the history of our code (e.g., Redis 2.6)?

File: redis.conf (hunk 1 above)
Comment (zuiderkwast) [reply redis.conf:176]: We discussed this in a core team meeting. Let's just remove the references to old Redis versions. (At least we can remove them *in most cases*.)

**Comment by zuiderkwast** [comment]
This is obsolete now. Fixed in other PRs.

**Pull request closed without merging**